Approve Screen Design QA

rickycodes opened the pull request:
<!--
Thanks for your contribution!

Please ensure that any applicable requirements below are satisfied before submitting this pull request. This will help ensure a quick and efficient review cycle.
-->

**Description**

re: https://github.com/MetaMask/metamask-mobile/issues/1330

**Approve**

![image](https://user-images.githubusercontent.com/675259/75632774-9d145480-5bcd-11ea-81ae-740cbeba1d9c.png)

**Approve 2**

![image](https://user-images.githubusercontent.com/675259/75632784-a8678000-5bcd-11ea-8801-f1d3df3a34f4.png)

**Edit Transaction Fee**

![image](https://user-images.githubusercontent.com/675259/75632799-b1585180-5bcd-11ea-828d-086fe9259d17.png)

**Edit Transaction Fee 2**

![image](https://user-images.githubusercontent.com/675259/75632808-bc12e680-5bcd-11ea-836c-2c4fff09d57c.png)

**Edit Permission**

![image](https://user-images.githubusercontent.com/675259/75632851-0300dc00-5bce-11ea-8681-28821ff6da48.png)

I've also removed the `ApproveSuccess` screen

**Checklist**

* [x] There is a related GitHub issue
* [x] Tests are included if applicable
* [x] Any added code is fully documented (not applicable, nothing was added)
* [x] Use 24px padding (or as low as 22px for smaller screen) to the left and right of the screen.
* [x] This includes address bar and large cancel/approve buttons.
* [x] The modal window looks like it goes over 24px. 
* [x] Remove confirm page.
* [x] Include screen shots.

**Issue**

Resolves #1330

rickycodes committed:
Adjust padding

rickycodes committed:
Add padding for ActionModal

rickycodes committed:
Do not navigate to ApproveSuccessView

rickycodes committed:
Update tests

rickycodes committed:
Remove ApproveSuccessView

rickycodes committed:
Update tests

cjeria reviewed (approved):
LGTM

estebanmino commented on app/components/UI/ActionModal/index.js:
```diff
@@ -14,10 +14,11 @@ const styles = StyleSheet.create({
 	},
 	modalContainer: {
-		width: '90%',
+		width: '100%',
```
this component is used in other places of the app not only here, is this working everywhere?

rickycodes replied:
yep, it's working as expected everywhere. I actually tested it via the `Send Feedback` link as well (since that's easier to get to).

estebanmino commented on app/components/UI/ActionView/index.js:
```diff
@@ -12,7 +12,8 @@ const styles = StyleSheet.create({
 		flexDirection: 'row',
-		padding: 16
+		paddingVertical: 16,
+		paddingHorizontal: 24
```
same here, component re used in other places

rickycodes replied:
we still want the same padding regardless of where it's used

estebanmino commented on app/components/UI/TransactionEdit/index.js:
```diff
@@ -43,8 +43,8 @@ const styles = StyleSheet.create({
-		right: 15,
-		left: 15,
+		right: 24,
+		left: 24,
```
this component is not related to approve screen, there are other updates as well?

rickycodes replied:
you're right that this one is not the in screens included in the PR, but the spacing on these still needed to be adjusted to match

estebanmino commented on app/components/Views/PaymentChannel/PaymentChannelDeposit/index.js:
```diff
@@ -138,7 +138,7 @@ const styles = StyleSheet.create({
 		borderRightWidth: 1,
 		paddingRight: 35,
-		paddingLeft: 20
+		paddingLeft: 24
```
same here, not approve

rickycodes replied:
the from graphic is in the screens and needed to be moved over slightly to fall into the same grid.

estebanmino reviewed (commented):
looks good, my only doubt is I see changes in other places that aren't related to approve screen

rickycodes commented:
I've addressed your feedback. there are a couple spacing changes that are unrelated to the approve screen, but this spacing should be the same throughout the application.

ibrahimtaveras00 commented:
I checked on various screen sizes on both OS's and the only issues I could fine are the following: 

Happening on a smaller screen (iPhone SE) but looks fine on the small screen on Nexus S

![Screen Shot 2020-03-04 at 3 09 04 PM](https://user-images.githubusercontent.com/41752922/75920938-49ed0c80-5e2e-11ea-9113-4887b2d54344.png)

The blue check mark on smaller screens:

Here's iPhone SE:

![Screen Shot 2020-03-04 at 3 30 08 PM](https://user-images.githubusercontent.com/41752922/75921029-6d17bc00-5e2e-11ea-9613-cbef0023fa24.png)

Here's Nexus S:

![Screen Shot 2020-03-04 at 3 37 38 PM](https://user-images.githubusercontent.com/41752922/75921051-76a12400-5e2e-11ea-980b-11fd2cdfa111.png)


And the "E" for "AVERAGE" on this view:

![Screen Shot 2020-03-04 at 3 36 10 PM](https://user-images.githubusercontent.com/41752922/75921131-97697980-5e2e-11ea-84c2-cb0f0dc7ebda.png)

which is happening on both iPhone SE and Nexus S

steps to get to this view:
- go to browser
- switch to ropsten network
- go to https://faucet.metamask.io/
- after connecting attempt to donate some ETH 
- on confirm view, tap on Edit on the top left

cjeria commented:
good catch @ibrahimtaveras00 !  I'll have a look at this asap!

cjeria commented:
Here are suggested solutions to all of the above UI bugs and link to figma file below. Let me know if there are any questions! cc @rickycodes 
![image](https://user-images.githubusercontent.com/597770/75941981-2abaa300-5e5f-11ea-8791-e178ad80eaec.png)

Figma file
https://www.figma.com/file/1aRs60GBYLLlBkReWg3XSA/Gas-selection-button-group?node-id=1%3A281

estebanmino reviewed (changes requested):
From yesterday's conversations, for this PR we have to address https://github.com/MetaMask/metamask-mobile/pull/1390#issuecomment-594825281 and apply @cjeria design in a separate PR

rickycodes commented:
Closing this and will open a new PR with this work as well as the new work suggested in https://github.com/MetaMask/metamask-mobile/pull/1390#issuecomment-594996341
